webboot/wget_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestWget(t *testing.T) {
+	type test struct {
+		name    string
+		isolink string
+		md5link string
+	}
+	tests := []test{
+		{name: "tinyCoreISO", isolink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", md5link: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso.md5.txt"},
+		{name: "archLinuxISO", isolink: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", md5link: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/md5sums.txt"},
+		{name: "ubuntuISO", isolink: "http://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso", md5link: "http://releases.ubuntu.com/bionic/MD5SUMS"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			r, err := wget(v.isolink)
+			if err != nil {
+				t.Errorf("Failed to receive the ISO: got %v, want nil", err)
+			}
+
+			body, err := ioutil.ReadAll(r)
+			if err != nil {
+				t.Errorf("Failed to read the ISO: got %v, want nil", err)
+			}
+			//computes the hash of the ISO & converts it into a string to compare with md5.txt hash
+			md := md5.Sum(body)
+			hash := hex.EncodeToString(md[:])
+
+			c, err := wget(v.md5link)
+
```

Comment:
I just noticed that. Thank you for pointing it out. However, a continue will not work here as well. Continue statements work in a for loop when you want to proceed with the rest of the code. Even if the continue statement worked, it would not logical because if getting the md5.txt file did not work, then you would not have an io.ReadCloser filled with contents to read and compare with the hash you generated from md5.Sum() for the OS you downloaded.

Reply:
In my opinion, I believe this would be Fatalf(). The user should not be able to continue past this if there is an error